src/uwtools/drivers/driver.py

```diff
@@ -174,43 +197,126 @@ def _taskname(self, suffix: str) -> str:
 
         :param suffix: Log-string suffix.
         """
-        return "%s %s" % (self._driver_name, suffix)
+        cycle = getattr(self, "_cycle", None)
+        leadtime = getattr(self, "_leadtime", None)
+        timestr = (
+            (cycle + leadtime).strftime("%Y%m%d %H:%M:%S")
+            if cycle and leadtime is not None
+            else cycle.strftime("%Y%m%d %HZ") if cycle else None
+        )
+        return " ".join(filter(None, [timestr, self._driver_name, suffix]))
```

Comment:
This version of `_taskname()` now flexibly deals with the time-invariant / cycle-based / cycle-and-leadtime-based drivers so that multiple implementations are not needed.